Classes/Command/OpenWeatherMapCommand.php

```diff
@@ -0,0 +1,217 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the package jweiland/weather2.
+ *
+ * For the full copyright and license information, please read the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace JWeiland\Weather2\Command;
+
+use JWeiland\Weather2\Utility\WeatherUtility;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use TYPO3\CMS\Core\Database\ConnectionPool;
+use TYPO3\CMS\Core\Http\RequestFactory;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3\CMS\Extbase\Service\CacheService;
+
+final class OpenWeatherMapCommand extends Command
+{
+    protected string $url = '';
+    protected string $dbExtTable = 'tx_weather2_domain_model_currentweather';
+    protected \stdClass $responseClass;
+    public string $clearCache = '';
+    public string $name = '';
+    public string $country = '';
+    public string $city = '';
+    public int $recordStoragePage = 0;
+
+    public function __construct(
+        private readonly LoggerInterface $logger,
+        private readonly RequestFactory $requestFactory,
+        private readonly CacheService $cacheService,
+        private readonly ConnectionPool $connectionPool,
+    ) {
+        parent::__construct();
+    }
+
+    protected function configure(): void
+    {
+        $this
+            ->setHelp('Calls the api of openweathermap.org and saves response into database')
+            ->addArgument('name', InputArgument::REQUIRED, 'Name')
+            ->addArgument('city', InputArgument::REQUIRED, 'City name (e.g. Munich)')
+            ->addArgument('country', InputArgument::REQUIRED, 'Country Code (e.g. DE)')
+            ->addArgument('apiKey', InputArgument::REQUIRED, 'API-Key')
+            ->addArgument(
+                'clearCache',
+                InputArgument::OPTIONAL,
+                'Clear cache for pages (comma separated list with IDs)',
+            )
+            ->addArgument('recordStoragePage', InputArgument::OPTIONAL, 'Record storage page (optional)');
+    }
+
+    public function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $output->writeln('<info>Starting OpenWeatherMap data fetch...</info>');
```

Comment:
Too much logic in command class. My idea was to move all that logic into its own service class. PLease interprete a command like a controller which is dump. Get data, and send it to somewhere else. No logic. That way we can call the service from everywhere and keep command clean and small